astropy/modeling/fitting.py

```diff
@@ -446,8 +461,18 @@ def __call__(self, model, x, y, z=None, weights=None, rcond=None):
                     # but z has a second axis for the model set. NB. This is
                     # ~5-10x faster than using rollaxis.
                     rhs = z.T if model_axis == 0 else z
+
+                if weights is not None:
+                    # Same for weights
+                    if weights.ndim > 2:
+                        weights = np.rollaxis(z, model_axis, z.ndim)
```

Comment:
Uh oh, this says `z` instead of `weights`!

Reply:
Oops, good catch, fixed.